core/test/sizes_test.cpp

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+#include <gtest/gtest.h>
+
+#include "test_macros.h"
+
+#include "scipp/core/sizes.h"
+
+using namespace scipp;
+
+using SmallMap =
+    scipp::core::small_stable_map<Dim, scipp::index, core::NDIM_MAX>;
+
+TEST(SmallMapTest, empty_size) {
```

Comment:
Might have been nice to have tested `SmallMap` independently. However as you say, if it's only (essentially) a detail `Sizes` this kind of testing is more valuable.